src/middlewares/auth/auth.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+export const authorization = (req, res, next) => {
+  const header = req.header('Authorization');
+
+  if (typeof header !== 'undefined') {
+    const bearer = header;
+    const token = bearer;
+    req.token = token;
+    next();
+  } else {
+    res.status(401).json({ status: 401, error: 'You are not authorized to access this route' });
+  }
+};
+
+export const jwtVerify = (req, res, next) => {
+  const { token } = req;
+  jwt.verify(token, process.env.SECRET_KEY_SIGN_UP, (err, result) => {
```

Comment:
Why secret key_sign_up? Are there different secrets for other routes? You can just say JWT_SECRET

Reply:
Yes they are different key for different routes, for example. For example the secret key to generate token sent for email verification is different for the secret key when a user signs up and token is giving. So we have two tokens. signup token and email token